### PR TITLE
Rabin Karp Algorithm implemented

### LINKED ISSUE #334: Rabin Karp Algorithm
**Algorithm: [Rabin-Karp Algorithm](https://en.wikipedia.org/wiki/Rabin%E2%80%93Karp_algorithm)**

**Language Supported:**
- C++ [@YashThakare ]
- C
- Java [@parjanyaacoder]
- Python [@AdityaJ7]
- C#
- Javascript
- Golang

### PR DESCRIPTION
**Pull request by parjanyaacoder**
Implemented Rabin Karp Algorithm which is a pattern searching algorithm in Java. 

- [x] I've checked the issue list before deciding what to submit.
- [x] I've read the contribution guidelines.
- [x] I've edited the `README.md` link to my code.

Fixes #334

### CODE REVIEW

File: Java/cp/Rabin_Karp.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import java.util.*;
```
Comment (atarax665) [inline Java/cp/Rabin_Karp.java:5]: Add Rabin-Karp algorithm's description.

File: Java/cp/Rabin_Karp.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    public static void main(String[] args)
+ {
+   String actualText,patternToBeSearched;
+   Scanner sc=new Scanner(System.in);
+   System.out.println("Enter actual text");
+   actualText=sc.nextLine();
+   System.out.println("Enter pattern to be searched");
+   patternToBeSearched=sc.nextLine();
+   int indexI,indexJ,locationIndex=-1,count=0;
```
Comment (atarax665) [inline Java/cp/Rabin_Karp.java]: Try following [Google java style guide](https://google.github.io/styleguide/javaguide.html) to write readable and clean code. For now, add space before and after each operator.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Java/cp/Rabin_Karp.java (hunk 1 above)
Comment (parjanyaacoder) [reply Java/cp/Rabin_Karp.java:5]: @atarax665 Please review.

**Review by atarax665 — APPROVED** [review]

**Comment by atarax665** [comment]
LGTM! Great work @parjanyaacoder

**Pull request merged**